api/src/test/java/org/openmrs/module/locationbasedaccess/aop/PatientSearchAdviserTest.java

```diff
@@ -0,0 +1,207 @@
+
+    @Override
+    protected void setInterceptorAndServices(TestWithAOP testCase) {
+        testCase.setInterceptor(PatientServiceInterceptorAdvice.class);
+        testCase.addService(PatientService.class);
+    }
+
+    @Before
+    public void setUp()  throws Exception {
+        userService = Context.getUserService();
+        patientService = Context.getPatientService();
+        locationService = Context.getLocationService();
+        executeDataSet(XML_FILENAME_WITH_ADMIN_USER_DATA);
+        executeDataSet(XML_FILENAME_WITH_PERSON_DATA);
+    }
+
+    @Test
+    public void getPatients_adminShouldGetPatientsFromAllLocations() {
+        Context.authenticate("admin", "test");
+        User authenticatedUser = Context.getAuthenticatedUser();
+        assertNotNull(authenticatedUser);
+        assertTrue(authenticatedUser.isSuperUser());
+
+        Location location1 = locationService.getLocation(1);
+        Location location2 = locationService.getLocation(2);
+
+        List<Patient> patientList = patientService.getPatients("Patient");
+        assertEquals(patientList.size(), 3);
+        assertEquals(patientList.get(0).getAttribute(10).getValue(), location1.getUuid());
+        assertEquals(patientList.get(1).getAttribute(10).getValue(), location2.getUuid());
+        assertEquals(patientList.get(2).getAttribute(10), null);
+    }
+
+    @Test
+    public void getPatient_adminShouldGetPatientsFromAllLocations() {
+        Context.authenticate("admin", "test");
+        User authenticatedUser = Context.getAuthenticatedUser();
+        assertNotNull(authenticatedUser);
+        assertTrue(authenticatedUser.isSuperUser());
+
+        Location location1 = locationService.getLocation(1);
+        Location location2 = locationService.getLocation(2);
+
+        Patient patient1 = patientService.getPatient(2);
+        assertNotNull(patient1);
+        assertEquals(patient1.getAttribute(10), null);
+
+        Patient patient2 = patientService.getPatient(3);
+        assertNotNull(patient2);
+        assertEquals(patient2.getAttribute(10).getValue(), location1.getUuid());
+
+        Patient patient3 = patientService.getPatient(4);
+        assertNotNull(patient3);
+        assertEquals(patient3.getAttribute(10).getValue(), location2.getUuid());
+    }
+
+    @Test
+    public void getPatientByUuid_adminShouldGetPatientsFromAllLocations() {
+        Context.authenticate("admin", "test");
+        User authenticatedUser = Context.getAuthenticatedUser();
+        assertNotNull(authenticatedUser);
+        assertTrue(authenticatedUser.isSuperUser());
+
+        Location location1 = locationService.getLocation(1);
+        Location location2 = locationService.getLocation(2);
+
+        Patient patient1 = patientService.getPatientByUuid("5631b434-78aa-102b-91a0-001e378eb67e");
+        assertNotNull(patient1);
+        assertEquals(patient1.getAttribute(10), null);
```

Comment:
When I changed to Id-1, it caused some unknown Test fails : https://pastebin.com/mLeh7QkP

Reply:
@dkayiwa Do you have any idea about this? 
expected:<NULL> but was:<null> for this line?

Reply:
What i mean is that when one looks at 10, it just looks like a magic number without meaning.

Reply:
Yes @dkayiwa  when I have `person_attribute_type_id=10 or above`, I could able to compile without any errors. But when I change it to `person_attribute_type_id=1`, I got those errors.

Reply:
You do not seem to be getting me. What does the magic number 10 or 1 mean?

Reply:
Ohh, Missed to get your point. 
The number 10 or 1 is referred to the value in the dummy XML data for the testing. See (here)[https://github.com/openmrs/openmrs-module-locationbasedaccess/pull/38/files#diff-5420cf0543307de782b1e37c5db47471R11]

Reply:
For someone looking at your code, how do they tell what the magic number 10 means?

Reply:
As the first step, I have executed the testData set (here)[https://github.com/openmrs/openmrs-module-locationbasedaccess/pull/38/files#diff-ba475a8a0e81f093e143aa265716d13bR52]. So it will point to that value.

Reply:
So do I want to provide some comments for those lines?

Reply:
ever read about self commenting code?

Reply:
Or read this: https://improvingsoftware.com/2011/06/27/5-best-practices-for-commenting-your-code/

Reply:
Got it, Make it as constant and I can fetch it from there. Thanks, @dkayiwa  for the wonderful tip !!!

Reply:
@dkayiwa I have updated the PR with the constants for the values.

Reply:
That's exactly what i wanted you to see for yourself without spoon feeding you. 😊

Reply:
I like your approach @dkayiwa  :smile: